src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -608,17 +608,23 @@ export class ResultsViewPageStore {
     }, []);
 
     readonly germlineConsentedSamples = remoteData<SampleIdentifier[]>({
+        await: () => [this.germlineSampleLists],
         invoke: async () => {
-            const studies: string[] = this.studyIds;
-            const ids: string[][] = await Promise.all(studies.map(studyId => {
-                return client.getAllSampleIdsInSampleListUsingGET({
-                    sampleListId: this.getGermlineSampleListId(studyId)
-                });
-            }));
-            return _.flatten(ids.map((sampleIds: string[], index: number) => {
-                const studyId = studies[index];
-                return sampleIds.map(sampleId => ({sampleId, studyId}));
-            }));
+            if (this.germlineSampleLists.result) {
+                const ids: string[][] = await Promise.all(this.germlineSampleLists.result.map(sampleList => {
+                    return client.getAllSampleIdsInSampleListUsingGET({
+                        sampleListId: sampleList.sampleListId
+                    });
+                }));
+
+                return _.flatten(ids.map((sampleIds: string[], index: number) => {
```

Comment:
this doesnt seem multiple-study-safe - need to perform conversion from sampleId to uniqueSampleKey, or just return Sample objects

Reply:
Does this mean that the current code is already not multiple-study-safe?

```JavaScript
return _.flatten(ids.map((sampleIds: string[], index: number) => {
    const studyId = studies[index];
    return sampleIds.map(sampleId => ({sampleId, studyId}));
}));
```

Reply:
nevermind, I didn't read carefully enough. looks good